databricks/koalas/internal.py

```diff
@@ -593,13 +595,15 @@ def __init__(
 
         data_dtypes = [
             spark_type_to_pandas_dtype(spark_frame.select(scol).schema[0].dataType)
-            if dtype is None
+            if dtype is None or dtype == np.dtype("object")
             else dtype
             for dtype, scol in zip(data_dtypes, data_spark_columns)
         ]
 
         assert all(
-            as_spark_type(dtype, raise_error=False) is not None for dtype in data_dtypes
+            isinstance(dtype, Dtype.__args__)  # type: ignore
```

Comment:
Can I take it as pandas dtype consists of np.dtype and ExtensionDtype?

Reply:
I believe so, so far. https://github.com/pandas-dev/pandas/blob/7af47c9/pandas/_typing.py#L126

Reply:
Got it, thanks!